### PR TITLE
Add Kafka Kerberos Configuration Properties

### PR DESCRIPTION
**Pull request by garyrussell**
Additional JAAS properties to support Kerberos.

### CODE REVIEW

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java [hunk 1]
```diff
@@ -776,4 +786,66 @@ public void setTruststorePassword(String truststorePassword) {
...
+		public boolean isEnabled() {
+			return this.enabled;
+		}
+
+		public void setEnabled(boolean enabled) {
+			this.enabled = enabled;
+		}
+
+		@NotNull
+		public String getLoginModule() {
```
Comment (snicoll) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java]: I am a bit confused. How is that supposed to work? If jaas support is enabled you don't want to make that a hard requirement do you? There is no test for that validation rule and we don't use the annotations ourselves anyway so I think we should go with dedicated logic for this.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java [hunk 2]
```diff
@@ -776,4 +786,66 @@ public void setTruststorePassword(String truststorePassword) {
 
 	}
 
+	@Validated
+	public class Jaas {
```
Comment (snicoll) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java]: missing `static`?

File: spring-boot-dependencies/pom.xml [hunk 3]
```diff
@@ -158,7 +158,7 @@
 		<spring-data-releasetrain.version>Kay-M3</spring-data-releasetrain.version>
 		<spring-hateoas.version>0.23.0.RELEASE</spring-hateoas.version>
 		<spring-integration.version>5.0.0.BUILD-SNAPSHOT</spring-integration.version>
-		<spring-kafka.version>2.0.0.M1</spring-kafka.version>
+		<spring-kafka.version>2.0.0.BUILD-SNAPSHOT</spring-kafka.version>
```
Comment (snicoll) [inline spring-boot-dependencies/pom.xml]: Can't merge this until we get M2 I suppose? We are releasing a milestone next week.

File: spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc [hunk 4]
```diff
@@ -944,6 +944,10 @@ content into your application; rather pick only the properties that you need.
 	spring.kafka.consumer.key-deserializer= # Deserializer class for keys.
 	spring.kafka.consumer.max-poll-records= # Maximum number of records returned in a single call to poll().
 	spring.kafka.consumer.value-deserializer= # Deserializer class for values.
+	spring.kafka.jaas.control-flag=REQUIRED # AppConfigurationEntry.LoginModuleControlFlag value.
+	spring.kafka.jaas.enabled= # Enable JAAS configuration.
```
Comment (snicoll) [inline spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc]: The documentation here _must_ be a copy/paste of the javadoc. This version is better than what you have in code.

**Comment by snicoll** [comment]
@garyrussell thanks for the PR. I guess you don't expect to release M2 anytime soon right? In that case we can't merge that PR before Spring Boot 2.0 M1 is out.

**Comment by garyrussell** [comment]
@snicoll M2 is scheduled to be released tomorrow - I will polish this PR according to your comments and update it after M2 is released.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java (hunk 2 above)
Comment (garyrussell) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java]: Oops

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java (hunk 1 above)
Comment (garyrussell) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java]: This is not needed; the assertions are done on the underlying object that is configured with these properties.

**Comment by garyrussell** [comment]
Note: this includes updating S-K to 2.0.0.M2.

If you don't want to include this in M1, a separate commit is needed for that update.

**Comment by snicoll** [comment]
I did create a separate issue for the M2 upgrade (see #9225).

Thanks @garyrussell !

**Pull request closed without merging**